wgengine/magicsock/magicsock.go

```diff
@@ -3186,6 +3292,37 @@ func (de *discoEndpoint) noteActiveLocked() {
 	}
 }
 
+// cliPing starts a ping for the "tailscale ping" command. res is value to call cb with,
+// already partially filled.
+func (de *discoEndpoint) cliPing(res *ipnstate.PingResult, cb func(*ipnstate.PingResult)) {
+	now := time.Now()
+
+	de.mu.Lock()
```

Comment:
defer the Unlock?

Reply:
I didn't because I didn't want to hold the mutex during the callback, but just launched the callback in a goroutine instead, like happens in the pong case anyway.